controllers/users_controller.js

```diff
@@ -3,133 +3,118 @@ const utils = require('../utils/util')
     }
     if (utils.isEmpty(email)) {
-        errorMessage += 'Email is Null\n'
+        errors.push('Email is null')
     }
     if (utils.isEmpty(name)) {
-        errorMessage += 'Name is Null\n'
+        errors.push('Name is null')
     }
 
-    if (!utils.isEmpty(errorMessage)) {
+    if (!utils.isEmpty(errors)) {
         res.status(401).json({
             error: true,
-            data: errorMessage
+            data: errors
         })
     }
     else {
-        user.register(username, password, email, name).subscribe(
-            id =>
-                res.status(201).json({
-                    error: false,
-                    data: id
-                }),
-            err => {
-                let message = '';
-                //noinspection EqualityComparisonWithCoercionJS
-                if (err.code == 23505) {
-                    message = `${utils.capitalize(err.detail.match(/[a-zA-Z]+(?=\))/)[0])} already exists`;
-                }
-                res.status(500).json({
+        try {
+            const {id, datecreated} = await user.register(username, password, email, name)
+            res.status(201).json({
+                error: false,
+                data: {name, username, email, datecreated, id}
+            })
+        }catch (e) {
+            if(e.code === '23505'){
+                return res.status(409).json({
                     error: true,
-                    data: message
+                    msg: `${utils.capitalize(e.detail.match(/[a-zA-Z]+(?=\))/)[0])} already exists`
                 })
-            })
+            }else{
+                console.log(e);
+                return res.status(500).json({error: true});
+            }
+        }
     }
 }
 
-module.exports.findUser = ({params: {username = null}}, res) => {
-    const observable = user.findUser(username);
-    observable.subscribe(
-        next => {
-            let [user, posts, followers, following] = next;
-            posts.map(post => {
-                post.user = {
-                    username: user.username,
-                    profile_picture_url: user.profile_picture_url,
-                    name: user.name
-                };
-                return post;
-            });
-            user.posts = posts;
-            user.followers = followers;
-            user.following = following;
-            res.status(202).json({
-                error: false,
-                data: user
-            })
-        },
-        err => {
-            res.status(400).json({
+module.exports.findUser = async ({params: {username = null}}, res) => {
+    try{
+        const [userInfo, postInfo = [], followersInfo = [], followingInfo = []] = await user.findUser(username)
+        postInfo.map(post => {
+            post.user = {
+                username: user.username,
+                profile_picture_url: user.profile_picture_url,
+                name: user.name
+            };
+            return post;
```

Comment:
no need to `return post;` as you don't catch the returned array:

`let newArr = postInfo.map(post => {}); // <-- case where you'd want to return "post"`

Reply:
Ahh yes, dumb mistake. I'll fix that